### PR TITLE
Add relocation support for simple raster states

### PR DESCRIPTION
**Pull request by csyonghe**
This change adds relocation for two simple rasterizer states: numSamples and samplePatternIdx.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JaxLinAMD** [comment]
please resolve conflicts

File: llpc/util/llpcElfWriter.cpp [hunk 1]
```diff
@@ -1209,6 +1209,42 @@ bool getDeviceIndexRelocationValue(Context *context, RelocationEntry relocEntry,
...
+// @param isGraphicsPipeline : Whether we are processing a compute or graphics pipeline
+// @param value : [out] The value of the relocation
+bool getNumSamplesRelocationValue(Context *context, RelocationEntry relocEntry, bool isGraphicsPipeline,
+                                  unsigned *value) {
+  auto pipelineBuildInfo = context->getPipelineContext()->getPipelineBuildInfo();
+  if (isGraphicsPipeline) {
+    *value = static_cast<const GraphicsPipelineBuildInfo *>(pipelineBuildInfo)->rsState.numSamples;
+  } else {
+    llvm_unreachable("num samples relocation is for graphics pipeline only.");
+  }
```
Comment (nhaehnle) [inline llpc/util/llpcElfWriter.cpp:1226]: Minor note: For stuff like this, you could just say `assert(isGraphicsPipeline)` or, if you want to be more verbose about it, `assert(isGraphicsPipeline && "num samples relocation is for graphics pipeline only");`. This is a common pattern in LLVM code.

File: llpc/util/llpcElfWriter.cpp [hunk 2]
```diff
@@ -1209,6 +1209,42 @@ bool getDeviceIndexRelocationValue(Context *context, RelocationEntry relocEntry,
...
+// @param isGraphicsPipeline : Whether we are processing a compute or graphics pipeline
+// @param value : [out] The value of the relocation
+bool getSamplePatternIdxRelocationValue(Context *context, RelocationEntry relocEntry, bool isGraphicsPipeline,
+                                        unsigned *value) {
+  auto pipelineBuildInfo = context->getPipelineContext()->getPipelineBuildInfo();
+  if (isGraphicsPipeline) {
+    *value = static_cast<const GraphicsPipelineBuildInfo *>(pipelineBuildInfo)->rsState.samplePatternIdx;
+  } else {
+    llvm_unreachable("samplesPatternIdx relocation is for graphics pipeline only.");
+  }
```
Comment (nhaehnle) [inline llpc/util/llpcElfWriter.cpp:1244]: The same applies here.

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3986738157c27e3668604981203a8bbb2779c6fa
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
@csyonghe please resolve conflicts

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit e469a9dfe18037f3ff11af96ee07d4b21b120ca2
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev b797a4faec565ad9a3ae987eb42fcf7c943203b1</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 53692d985a53a336e07907c2d4b86bf2deb66908</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 40826/40827 (100.0%)</li>
<li><details><summary>Failed: 1/40827 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.spirv_assembly.instruction.graphics.convertftou.float32_to_uint32_geom
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**